uaclient/cli.py

```diff
@@ -51,15 +52,27 @@
 STATUS_FORMATS = ["tabular", "json"]
 
 
-def assert_attached_root(func):
-    """Decorator asserting root user and attached config."""
+def assert_attached_root(unattached_msg_tmpl=None):
+    """Decorator asserting root user and attached config.
 
-    def wrapper(args, cfg):
-        if os.getuid() != 0:
-            raise exceptions.NonRootUserError()
-        if not cfg.is_attached:
-            raise exceptions.UnattachedError()
-        return func(args, cfg)
+    :param unattached_msg_tmpl: Optional msg template to format if raising an
+        UnattachedError
+    """
+
+    def wrapper(f):
```

Comment:
This function is the decorator, `decorator` below might be better as `new_f` or similar?